src/components/FolderCard.jsx

```diff
@@ -24,10 +28,18 @@ const FolderCard = ({
   const [isFolderModalOpen, setIsFolderModalOpen] = useState(false)
   const [canShowDropdown, setCanShowDropdown] = useState(false)
   const [canShowDeleteWarningModal, setCanShowDeleteWarningModal] = useState(false)
+  const [isOutOfViewport, setIsOutOfViewport] = useState()
   const dropdownRef = useRef(null)
 
   useEffect(() => {
-    if (canShowDropdown) dropdownRef.current.focus()
+    if (canShowDropdown) {
+      dropdownRef.current.focus()
+      if (isOutOfViewport === undefined) {
+        // We only want to run this once
+        const bounding = dropdownRef.current.getBoundingClientRect()
+        setIsOutOfViewport(checkIsOutOfViewport(bounding, ['right']))
+      }
+    }
```

Comment:
Nit: currently, since this is calculated only after dropdown is shown, we get a slight flicker where the dropdown for a card on the extreme right is shown to the right before it's set to the left of the card. Ideally, we would be able to calculate which cards are at the right edge and set the dropdown modal to appear on the left instead

Reply:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/247/commits/5712fe6212b2ad2cd2e847c640ea94c2d9d2ae22